### PR TITLE
Added Validating and Signing

### PR DESCRIPTION
**Pull request by OttoVT**

### CODE REVIEW

File: src/Brokerage.Common/Brokerage.Common.csproj [hunk 1]
```diff
@@ -37,8 +37,8 @@
     <PackageReference Include="Swisschain.Extensions.Idempotency.EfCore" Version="2.1.0" />
     <PackageReference Include="Swisschain.Extensions.Idempotency.MassTransit" Version="3.0.1" />
     <PackageReference Include="Swisschain.Sirius.Confirmator.MessagingContract" Version="3.0.1-beta" />
-    <PackageReference Include="Swisschain.Sirius.Executor.ApiClient" Version="2.4.0" />
-    <PackageReference Include="Swisschain.Sirius.Executor.MessagingContract" Version="1.3.0" />
+    <PackageReference Include="Swisschain.Sirius.Executor.ApiClient" Version="2.5.0-beta.1" />
```
Comment (KonstantinRyazantsev) [inline src/Brokerage.Common/Brokerage.Common.csproj:40]: release it

File: src/Brokerage.Common/Brokerage.Common.csproj [hunk 2]
```diff
@@ -37,8 +37,8 @@
...
+    <PackageReference Include="Swisschain.Sirius.Executor.MessagingContract" Version="1.4.0-beta.2" />
```
Comment (KonstantinRyazantsev) [inline src/Brokerage.Common/Brokerage.Common.csproj:41]: release it

File: src/Brokerage.Common/Domain/Processing/ISigningOperationProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+using Swisschain.Sirius.Executor.MessagingContract;
+
+namespace Brokerage.Common.Domain.Processing
+{
+    public interface ISigningOperationProcessor
+    {
+        Task Process(OperationSigning evt, OperationProcessingContext processingContext);
+    }
+
+    public interface IValidatingOperationProcessor
```
Comment (KonstantinRyazantsev) [inline src/Brokerage.Common/Domain/Processing/ISigningOperationProcessor.cs:12]: several types in the same file

File: src/Brokerage.Common/Domain/Processing/IProcessorsFactory.cs [hunk 4]
```diff
@@ -7,6 +7,8 @@ public interface IProcessorsFactory
         IReadOnlyCollection<IDetectedTransactionProcessor> GetDetectedTransactionProcessors();
         IReadOnlyCollection<IConfirmedTransactionProcessor> GetConfirmedTransactionProcessors();
         IReadOnlyCollection<ISentOperationProcessor> GetSentOperationProcessors();
+
+        IReadOnlyCollection<ISigningOperationProcessor> GetSigningOperationProcessors();
```
Comment (KonstantinRyazantsev) [inline src/Brokerage.Common/Domain/Processing/IProcessorsFactory.cs:11]: and what about validating operations processors?

File: src/Brokerage.Common/Domain/Withdrawals/Processors/SigningWithdrawalProcessor.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            foreach (var withdrawal in processingContext.Withdrawals)
+            {
+                withdrawal.MoveToSigning();
+            }
+
+            return Task.CompletedTask;
+        }
+    }
+
+    //public class ValidatinWithdrawalProcessor : IValidatingOperationProcessor
```
Comment (KonstantinRyazantsev) [inline src/Brokerage.Common/Domain/Withdrawals/Processors/SigningWithdrawalProcessor.cs:27]: commented code

File: src/Brokerage.Common/Domain/Withdrawals/Withdrawal.cs [hunk 6]
```diff
@@ -190,9 +190,29 @@ public void Complete()
...
+
+            UpdatedAt = DateTime.UtcNow;
+
+            AddUpdateEvent();
+        }
+
+        //TODO: Use it in the future for executing state
+        public void MoveToExecuting()
+        {
+            SwitchState(new[] { WithdrawalState.Signing }, WithdrawalState.Executing);
```
Comment (KonstantinRyazantsev) [inline src/Brokerage.Common/Domain/Withdrawals/Withdrawal.cs:205]: Let's remove Executing at all. I've updated diagram as well

File: src/Brokerage.Worker/Brokerage.Worker.csproj [hunk 7]
```diff
@@ -24,7 +24,8 @@
     <PackageReference Include="Swisschain.Extensions.MassTransit" Version="3.0.4" />
     <PackageReference Include="swisschain.extensions.postgres" Version="1.0.0" />
     <PackageReference Include="Swisschain.Sdk.Server" Version="6.8.0" />
-    <PackageReference Include="Swisschain.Sirius.Executor.ApiClient" Version="2.4.0" />
+    <PackageReference Include="Swisschain.Sirius.Executor.ApiClient" Version="2.5.0-beta.1" />
```
Comment (KonstantinRyazantsev) [inline src/Brokerage.Worker/Brokerage.Worker.csproj:27]: Release it

File: src/Brokerage.Worker/Brokerage.Worker.csproj [hunk 8]
```diff
@@ -24,7 +24,8 @@
...
+    <PackageReference Include="Swisschain.Sirius.Executor.MessagingContract" Version="1.4.0-beta.2" />
```
Comment (KonstantinRyazantsev) [inline src/Brokerage.Worker/Brokerage.Worker.csproj:28]: Release it

File: src/Brokerage.Common/Domain/Processing/IProcessorsFactory.cs (hunk 4 above)
Comment (OttoVT) [reply src/Brokerage.Common/Domain/Processing/IProcessorsFactory.cs:11]: We dont need it

**Review by KonstantinRyazantsev — APPROVED** [review]

**Pull request merged**